browser/src/Services/Explorer/ExplorerFileSystem.ts

```diff
@@ -58,4 +80,89 @@ export class FileSystem implements IFileSystem {
     public exists(fullPath: string): Promise<boolean> {
         return this._fs.exists(fullPath)
     }
+
+    /**
+     * Delete a file or Folder
+     *
+     * @name deleteNode
+     * @function
+     * @param {ExplorerNode} node The file or folder node
+     */
+    public deleteNode = async (node: ExplorerNode): Promise<void> => {
+        switch (node.type) {
+            case "folder":
+                await remove(node.folderPath)
```

Comment:
fs-extra dosen't let us to set the inner fs module used for it (Seems to be a pr pending https://github.com/jprichardson/node-fs-extra/pull/558), so we are always removing file on disk ignoring the fs arg, it was why I was sugesting rimraf because it let us specify the fs instance to use

For mv however i don't find a good module for it, but it could be implemented pretty easily if we upgrade electron (Give us fs.copyFile) https://github.com/sindresorhus/move-file/blob/master/index.js#L9-L34

Reply:
Thanks for the feedback, I actually had a look at alternatives like `rimraf` and co initially, but *decided* to go with `fs-extra` because not only is it a well-typed, maintained package, it gives us all of `fs` functionality already promisified, adds on approx 12 new v. useful `async` methods not just for this PR but for upcoming changes like `rename`, `add` and the filesystem watcher related functionality.

I think a green PR on the v. active repo is a good sign that they will implement this functionality so we aren't likely to be missing this for long, vs. the trade off using several different packages to patch together the same functionality that this *one* provides.

Also a big goal of this PR is to get people to use the `persistable delete` I implemented which means actually moving files to the tmp dir (possibly configurable later) as opposed to deleting them from the drive hopefully the actually delete would be used only when users really want a file gone in which case I think it is valid to have it actually removed from the system.

Tbh switching packages from `shelljs` was actually not trivial and led to an almost complete rewrite of this PR because of redux-observable and how it handles promises and while any further changes might not be that big i'm looking to close this out once I've added more tests so if it feels necessary later to change things then that can come when/if a change seems warranted.